src/core/thread/mesh_forwarder.cpp

```diff
@@ -1481,7 +1507,7 @@ void MeshForwarder::HandleReassemblyTimer(void)
 
     if (mReassemblyList.GetHead() != NULL)
     {
-        mReassemblyTimer.Start(kStateUpdatePeriod);
+        mUpdateTimer.Start(kStateUpdatePeriod);
```

Comment:
How about having `UpdateReassemblyList()` and `UpdateFragmentLifetime()` return a `bool` to indicate whether or not to keep the timer running and have a single call to `mUpdateTimer.Start()` in `HandleUpdateTimer()`?

Reply:
Changed.